src/components/MenuItem.js

```diff
@@ -17,7 +17,7 @@ const propTypes = {
     onPress: PropTypes.func.isRequired,
 
     // Icon to display on the left side of component
-    icon: PropTypes.func.isRequired,
+    icon: PropTypes.elementType.isRequired,
```

Comment:
I've never seen `elementType` used before, and I'm a little unclear on how it's supposed to be used. The only hint I can find in the React docs is:

```
// A React element type (ie. MyComponent).
```

Which doesn't make a whole lot of sense... (I would expect to see "MyComponent" in their example somewhere). Ref: https://reactjs.org/docs/typechecking-with-proptypes.html#proptypes
but for this particular instance, I think `PropTypes.node` might be a better propType?

Reply:
`PropType.elementType` stands for either a stateless func component or a class component constructor e.g. not just any function but a specifically a react component.

It does sound a bit confusing but it exactly represents the requirement.

PropType.node would raise a warning as it stands for an evaluated/rendered content e.g. it's used when you have a prop like this:
```jsx
 menuItems={[
   {
     icon: <Paperclip />,
     text: 'Upload Photo',
     onSelect: () => {},
```

While with `PropType.elementType` the signature is this:

```jsx
 menuItems={[
   {
     icon: Paperclip,
     text: 'Upload Photo',
     onSelect: () => {},
```

The latter would allow you to easily pass more props to `Paperclip` if needed e.g. fill, stroke, other svg props

Reply:
OK, neat! Thanks for explaining that subtle difference. Let's keep it `element` in that case!

Reply:
Here I've followed this to the actual usage. It's used like an element:

https://github.com/Expensify/Expensify.cash/blob/d3f1d28478357e2a66b155e660def191d6ed79a8/src/components/Icon/index.js#L31-L38